src/processes/auto-welcome-msg-process.js

```diff
@@ -37,9 +40,27 @@ const automateWelcomeMsgProcess = async (job, done) => {
 
         await chat.sendMessage({
           ...toBeSent,
-          user_id: resultPrepopulated.targetUser.user_id
+          user_id: resultPrepopulated.targetUser.user_id,
+          skip_push: true
         });
 
+        // send to multiple token
+        const userTargetTokens = await FcmTokenFunction.findAllTokenByUserId(
+          FcmToken,
+          resultPrepopulated.targetUser.user_id
+        );
+
+        await Promise.all(
+          userTargetTokens.map(async (user) => {
+            const payload = {
+              notification: {
+                body: toBeSent
+              }
+            };
+            await messaging().sendToDevice(user?.token, payload);
+          })
+        );
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues in this section:

1. Error handling: There is no error handling for the `FcmTokenFunction.findAllTokenByUserId` function and the `messaging().sendToDevice` function. If any of these functions fail, it could cause the entire process to crash. It would be better to wrap these calls in a try-catch block.

2. Performance: The use of `Promise.all` with async functions can lead to unhandled promise rejections if one of the promises fails. Instead, you should use `Promise.allSettled`, which waits for all promises to settle, regardless of whether they have been fulfilled or rejected.

3. Data validation: There's no check to ensure that `user?.token` exists before trying to send a message to it. This could potentially lead to errors if a user doesn't have a token.

Here's how you might address these issues:

```diff
-        const userTargetTokens = await FcmTokenFunction.findAllTokenByUserId(
-          FcmToken,
-          resultPrepopulated.targetUser.user_id
-        );
-
-        await Promise.all(
-          userTargetTokens.map(async (user) => {
-            const payload = {
-              notification: {
-                body: toBeSent
-              }
-            };
-            await messaging().sendToDevice(user?.token, payload);
-          })
-        );

+        try {
+          const userTargetTokens = await FcmTokenFunction.findAllTokenByUserId(
+            FcmToken,
+            resultPrepopulated.targetUser.user_id
+          );
+
+          const notifications = userTargetTokens.map((user) => {
+            if (user?.token) {
+              const payload = {
+                notification: {
+                  body: toBeSent
+                }
+              };
+              return messaging().sendToDevice(user.token, payload);
+            }
+          }).filter(Boolean); // remove undefined values
+
+          await Promise.allSettled(notifications);
+        } catch (error) {
+          console.error('Failed to send notifications:', error);
+        }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->